src/bio/bio_xstream.c

```diff
@@ -380,48 +155,19 @@ bio_nvme_init(const char *nvme_conf, int shm_id, int mem_size,
 		return 0;
 	}
 
-	fd = open(nvme_conf, O_RDONLY, 0600);
-	if (fd < 0) {
-		D_WARN("Open %s failed, skip DAOS NVMe setup "DF_RC"\n",
-		       nvme_conf, DP_RC(daos_errno2der(errno)));
-		nvme_glb.bd_nvme_conf = NULL;
-		return 0;
-	}
-	close(fd);
-
 	rc = smd_init(db);
```

Comment:
It's not from this patch, but error cleanup is missed if smd_init() fails. Also, the order of error cleanup code below is wrong. (I suppose the smd_init() was the first call then moved to the middle by some commit, but the error cleanup code wasn't adjusted).

Reply:
good spot, thanks